openfeature/client_test.go

```diff
@@ -749,6 +749,87 @@ func TestRequirement_1_4_13(t *testing.T) {
 // Is tested by TestRequirement_4_4_2.
 
 // TODO Requirement_1_6_1
+
+func TestWaitForProvider(t *testing.T) {
```

Comment:
It might be nice to add a test that works like this:

- set a provider which is ready after 1s (has an init function that returns after 1s)
- returns a valid flag value after that time only
- returns an error before 1s

This is consistent with how we'd expect many providers to work. Maybe you could add some options to `NewMockFeatureProvider` that would support this?

Reply:
okay, I got the idea. I'll give it a try, but since I'm new to Go, it might take me some time 😇 .

Reply:
I just had a look at the test, and to make sure the `Init` function actually gets called, you will need to have a type that consolidates a `MockFeatureProvider` and `MockStateHandler` - otherwise the check [here](https://github.com/open-feature/go-sdk/pull/244/files#diff-434e8d62e6a56910999195bb04f7c78e5373efd76288dc4039cd58686c99ae8bR270) will fail and the function will proceed without calling the `Init` function of the provider.

Since gomock, which we use for creating the mocks seems to not support the generation of a type that implements multiple interfaces, i believe the simplest way to get a struct that implements both the FeatureProvider and the StateHandler interfaces is to consolidate them into a single struct, such as follows:

```go
type mockProviderAndStateHandler struct {
	*MockFeatureProvider
	*MockStateHandler
}
```

This type should be defined in the `client_test.go` file, and can then be used to pass it to the `SetProviderAndWait` function, such as in the following example:

```
...
		mockProvider.EXPECT().Metadata().AnyTimes()
		mockProvider.EXPECT().Hooks().AnyTimes()
		mockProvider.EXPECT().IntEvaluation(context.Background(), flagKey, defaultValue, flatCtx).
			Return(IntResolutionDetail{
				Value: 0,
				ProviderResolutionDetail: ProviderResolutionDetail{
					ResolutionError: NewGeneralResolutionError("test"),
				},
			}).Times(2)

		mockStateHandler := NewMockStateHandler(ctrl)

                // TODO add the desired behavior for the Init function of the MockStateHandler

		o := & mockProviderAndStateHandler{
			MockFeatureProvider: mockProvider,
			MockStateHandler:    mockStateHandler,
		}

		err := SetProviderAndWait(o)
...
```